File: chata_ai/src/main/java/chata/can/chata_ai/request/RequestApi.java
package chata.can.chata_ai.request;

import org.json.JSONObject;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.net.HttpURLConnection;
import java.net.URL;

public class RequestApi
{
	public RequestApi()
	{
		callRequest("https://carlos-buruel.000webhostapp.com/test_login.php");
		//callRequest("http://backend.chata.ai/oauth/token");
	}

	void callRequest(String urlPath)
	{
		try
		{
			URL url = new URL(urlPath);
			final HttpURLConnection connection = (HttpURLConnection) url.openConnection();
			connection.setConnectTimeout(5000);
			connection.setReadTimeout(5000);

			//connection.setRequestProperty("Content-Type", "application/json");
			connection.setDoOutput(true);
			connection.setRequestMethod("POST");

			final JSONObject json = new JSONObject();
			json.put("scope", "read");
			json.put("grant_type", "password");
			json.put("username", "carlos@rinro.com.mx");
			json.put("password", "$Chata124");

			Thread thread = new Thread(new Runnable()
			{
				@Override
				public void run()
				{
					try {
						OutputStreamWriter outputStreamWriter = new OutputStreamWriter(connection.getOutputStream());
						outputStreamWriter.write(json.toString());
						outputStreamWriter.close();

						InputStreamReader inputStreamReader = new InputStreamReader(connection.getInputStream());
						int codeResponse = connection.getResponseCode();

						if (codeResponse == 200)
						{
							BufferedReader reader = new BufferedReader(inputStreamReader);
							StringBuilder out = new StringBuilder();

							for (String line; (line = reader.readLine()) != null;)
							{
								out.append(line);
							}
							System.out.println("Code " + codeResponse);
							System.out.println("Response " + out);
						}
						else
						{
							System.out.println("Code " + codeResponse);
						}
					}
					catch (Exception ex)
					{
						ex.printStackTrace();
					}
				}
			});
			thread.start();
		}
		catch (Exception ex)
		{

		}
	}
}

File: chata_ai/src/main/java/chata/can/chata_ai/request/RequestApi.kt
package chata.can.chata_ai.request

import android.util.Log
import org.json.JSONObject
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.net.HttpURLConnection
import java.net.MalformedURLException
import java.net.ProtocolException
import java.net.URL

object RequestApi
{
	fun callStringRequest(pathUrl: String)
	{
		try
		{
			val url = URL(pathUrl)
			val connection = url.openConnection() as HttpURLConnection
			connection.connectTimeout = 5000
			connection.readTimeout = 5000
			connection.setRequestProperty("Content-Type", "application/json")
			connection.doOutput = true
			connection.requestMethod = "POST"

			val jsonTest = JSONObject()
			jsonTest.put("scope" , "read")
			jsonTest.put("grant_type", "password")
			jsonTest.put("username", "carlos@stx.com.mx")
			jsonTest.put("password", "\$Chata124")

			Thread(Runnable {
				val output = OutputStreamWriter(connection.outputStream)
				output.write(jsonTest.toString())
				output.close()

				val inputStreamReader = InputStreamReader(connection.inputStream)
				connection.responseCode

				//val allText = connection.inputStream.bufferedReader().use(BufferedReader::readText)

				val reader = BufferedReader(inputStreamReader)
				val sOut = ""
			}).start()
		}
		catch(ex: MalformedURLException)
		{
			Log.e("ERROR", "MalformedURLException: ${ex.message ?: ""}")
		}
		catch(ex: ProtocolException)
		{
			Log.e("ERROR", "ProtocolException: ${ex.message ?: ""}")
		}
		catch (ex: IOException)
		{
			Log.e("ERROR", "IOException: ${ex.message ?: ""}")
		}
		catch (ex: Exception)
		{
			Log.e("ERROR", "Exception: ${ex.message ?: ""}")
		}
		finally
		{

		}
	}
}